pkg/controllers/cluster/kubeconfig_controller.go

```diff
@@ -0,0 +1,257 @@
+		Watches(&corev1.Secret{}, handler.EnqueueRequestsFromMapFunc(clusterSecretToCluster)).
+		Watches(&v1alpha1.Organization{}, handler.EnqueueRequestsFromMapFunc(r.organizationToClusters)).
+		Watches(&corev1.Secret{}, handler.EnqueueRequestsFromMapFunc(r.organizationSecretToClusters)).
+		For(&v1alpha1.ClusterKubeconfig{}).
+		Complete(r)
+}
+
+func (r *KubeconfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	l := log.FromContext(ctx).WithValues("cluster", req.Name, "namespace", req.Namespace)
+
+	var cluster v1alpha1.Cluster
+	if err := r.Get(ctx, req.NamespacedName, &cluster); err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			return ctrl.Result{}, err
+		}
+		l.Info("skip reconcile, cluster is not found")
+		return ctrl.Result{}, nil
+	}
+
+	if cluster.GetDeletionTimestamp() != nil {
+		l.Info("skip reconcile, cluster is being deleted")
+		return ctrl.Result{}, nil
+	}
+
+	var kubeconfig v1alpha1.ClusterKubeconfig
+
+	if err := r.Get(ctx, req.NamespacedName, &kubeconfig); err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			return ctrl.Result{}, err
+		}
+
+		l.Info("kubeconfig not found, will be created")
+		kubeconfig.Name = cluster.Name
+		kubeconfig.Namespace = cluster.Namespace
+		kubeconfig.Spec.Kubeconfig = v1alpha1.ClusterKubeconfigData{
+			Kind:       "Config",
+			APIVersion: "v1",
+		}
+		kubeconfig.OwnerReferences = []metav1.OwnerReference{
+			{
+				APIVersion: cluster.APIVersion,
+				Kind:       cluster.Kind,
+				Name:       cluster.Name,
+				UID:        cluster.UID,
+			},
+		}
+		kubeconfig.Spec.Kubeconfig.Contexts = []v1alpha1.ClusterKubeconfigContextItem{
+			{
+				Name: cluster.Name,
+				Context: v1alpha1.ClusterKubeconfigContext{
+					Cluster:   cluster.Name,
+					AuthInfo:  "oidc@" + cluster.Name,
+					Namespace: "default",
+				},
+			}}
+		kubeconfig.Spec.Kubeconfig.CurrentContext = cluster.Name
+		kubeconfig.Status.Conditions.SetConditions(v1alpha1.TrueCondition(v1alpha1.KubeconfigCreatedCondition, "NewCreation", ""))
+	}
+
+	// get oidc info from organization
+	oidc, err := r.getOIDCInfo(ctx, cluster.Namespace)
+	if err != nil {
+		l.Error(err, "skip reconcile, oidc data not fetched")
+		return ctrl.Result{}, nil
+	}
+
+	// get cluster connection data from cluster secret
+	var secret corev1.Secret
+	err = r.Get(ctx, client.ObjectKey{Name: req.Name, Namespace: req.Namespace}, &secret)
+	if err != nil {
+		l.Error(err, "skip reconcile, secret data not fetched")
+		return ctrl.Result{}, nil
+	}
+
+	// collect cluster connection data and update kubeconfig
+	rootKubeCfg := secret.Data["kubeconfig"]
+	kubeCfg, err := clientcmd.Load(rootKubeCfg)
+	if err != nil {
+		l.Error(err, "unable to load kubeconfig")
+		return ctrl.Result{}, err
+	}
+
+	var clusterCfg *clientcmdapi.Cluster
+	for _, v := range kubeCfg.Clusters {
+		clusterCfg = v
+		break
+	}
+
+	// collect oidc data and update kubeconfig
+	clientutil.CreateOrPatch(ctx, r.Client, &kubeconfig, func() error {
```

Comment:
Can we log the result of this `CreateOrPatch` as in [here](https://github.com/cloudoperators/greenhouse/blob/main/pkg/controllers/cluster/bootstrap_controller.go#L170-L173)

Reply:
fixed by 16b7fb7af545499f90a2f168752ee2ceb335a60a